WordPress/src/main/java/org/wordpress/android/ui/notifications/utils/FormattableContentClickHandler.kt

```diff
@@ -0,0 +1,133 @@
+                            id)
+                } else {
+                    showWebViewActivityForUrl(activity, clickedSpan.url)
+                }
+            }
+            FormattableRangeType.STAT, FormattableRangeType.FOLLOW ->
+                // We can open native stats if the site is a wpcom or Jetpack sites
+                showStatsActivityForSite(activity, siteId, rangeType)
+            FormattableRangeType.LIKE -> if (ReaderPostTable.postExists(siteId, id)) {
+                showReaderPostLikeUsers(activity, siteId, id)
+            } else {
+                showPostActivity(activity, siteId, id)
+            }
+            else ->
+                // We don't know what type of id this is, let's see if it has a URL and push a webview
+                if (!TextUtils.isEmpty(clickedSpan.url)) {
+                    showWebViewActivityForUrl(activity, clickedSpan.url)
+                }
+        }
+    }
+
+    private fun showBlogPreviewActivity(activity: FragmentActivity, siteId: Long) {
+        ReaderActivityLauncher.showReaderBlogPreview(activity, siteId)
+    }
+
+    private fun showPostActivity(activity: FragmentActivity, siteId: Long, postId: Long) {
+        ReaderActivityLauncher.showReaderPostDetail(activity, siteId, postId)
+    }
+
+    private fun showStatsActivityForSite(activity: FragmentActivity, siteId: Long, rangeType: FormattableRangeType) {
+        val site = siteStore.getSiteBySiteId(siteId)
+        if (site == null) {
+            // One way the site can be null: new site created, receive a notification from this site,
+            // but the site list is not yet updated in the app.
+            ToastUtils.showToast(activity, R.string.blog_not_found)
+            return
+        }
+        showStatsActivityForSite(activity, site, rangeType)
+    }
+
+    private fun showStatsActivityForSite(
+        activity: FragmentActivity,
+        site: SiteModel,
+        rangeType: FormattableRangeType
+    ) {
+        if (rangeType == FormattableRangeType.FOLLOW) {
+            val intent = Intent(activity, StatsViewAllActivity::class.java)
+            intent.putExtra(StatsAbstractFragment.ARGS_VIEW_TYPE, StatsViewType.FOLLOWERS)
+            intent.putExtra(StatsAbstractFragment.ARGS_TIMEFRAME, StatsTimeframe.DAY)
+            intent.putExtra(StatsAbstractFragment.ARGS_SELECTED_DATE, "")
+            intent.putExtra(StatsAbstractFragment.ARGS_IS_SINGLE_VIEW, true)
+            intent.putExtra(StatsActivity.ARG_LOCAL_TABLE_SITE_ID, site.id)
+
+            intent.putExtra(
+                    StatsViewAllActivity.ARG_STATS_VIEW_ALL_TITLE,
+                    activity.getString(R.string.stats_view_followers)
+            )
+            activity.startActivity(intent)
+        } else {
+            ActivityLauncher.viewBlogStats(activity, site)
+        }
+    }
+
+    private fun showWebViewActivityForUrl(activity: FragmentActivity, url: String?) {
+        if (url == null) {
```

Comment:
We may want to consider adding a log message here or making the url parameter NonNull (and add the log message into the `onClick` method). If you decide to add a log message here, I'd suggest removing the not empty check on line 66 and ignore my previous comment.

Reply:
I did the second thing, I'm moving the null check here and passing range type as a parameter (to have it logged)